examples/Example.hs

```diff
@@ -4,35 +4,69 @@
 module Main where
 
 import qualified Data.Aeson as J
-import qualified Data.ByteString.Lazy as BL
+import qualified Data.Aeson.Key as K
+import qualified Data.Aeson.KeyMap as K
+import Data.Bifunctor (first)
 import Data.Maybe
 import Data.Text (Text)
-import Kriti (runKriti)
+import qualified Data.Text as T
+import Kriti (renderPretty, runKritiWith)
+import Kriti.Error (CustomFunctionError (..))
 import Network.HTTP.Client
+import Network.HTTP.Client.TLS
+import Prettyprinter
 import Text.RawString.QQ
 
 exampleTemplate :: Text
 exampleTemplate =
   [r|
-{
-   'id': {{$.id}},
-   'name': {{$.name.english}},
-   'description': {{$.description}},
-   'profile': {
-       'height': {{$.profile.height}},
-       'weight': {{$.profile.weight}},
-       'hp': {{$.base.HP}},
-       'attack': {{$.base.Attack}},
-       'defence': {{$.base.Defense}},
-       'speed': {{$.base.Speed}}
-   }
-}
+{{ range i, x := $.results }}
+  {
+    "id" : {{i}},
+    "fullName" : {{concatName x.name}},
+    "profile" : {
+      "gender" : {{getG x.gender}},
+      "emailID": {{x.email}},
+      "isSuperUser" : {{isAdmin x.login.username}}
+    }
+  }
+{{ end }}
+
 |]
 
 main :: IO ()
 main = do
-  manager <- newManager defaultManagerSettings
-  request <- parseRequest "http://app.pokemon-api.xyz/pokemon/pikachu"
+  manager <- newManager tlsManagerSettings
+  request <- parseRequest "https://randomuser.me/api/?results=10"
```

Comment:
I think it would make sense to actually avoid making network calls when testing Kriti since it could be tested equivalently with files, or literals.

Reply:
This would avoid network errors, timeouts, traffic, changing apis, etc. as well as making it easier to see the input data.

Reply:
Sure, I will add a `source.json`.
FYI, this is just an example 🙈